### PR TITLE
Create settings screen general tab

### LINKED ISSUE #497: [UI] Implement settings screen general tab
- [x] Create `GeneralSettings` component
  - [x] Add app name input
  - [ ] ~~Add app logo preview and link input~~
  - [x] Add save button
- [ ] Validate app name input
- [ ] Integrate query
- [ ] Integrate mutation
- [x] Block navigation if there are unsaved changes

### PR DESCRIPTION
**Pull request by nicolaslee2775**
close #497 

image

### CODE REVIEW

**Comment by rickmak** [comment]
hi @nicolaslee2775, please assign to @louischan-oursky explicitly.

**Review by axolotl0212 — APPROVED** [review]

File: portal/src/graphql/portal/GeneralSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }, []);
+
+  return (
+    <form className={styles.root} onSubmit={onFormSubmit}>
+      <TextField
+        className={styles.textField}
+        label={renderToString("GeneralSettings.app-name.label")}
+        value={state.appName}
+        onChange={onAppNameChange}
+      />
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/GeneralSettings.tsx:54]: For now, can we just show the app ID in a disabled text field? So the save button can also be always disabled.

Comment (nicolaslee2775) [reply portal/src/graphql/portal/GeneralSettings.tsx:54]: ok

**Comment by nicolaslee2775** [comment]
@louischan-oursky Rebased and replaced app name with app ID.

image

**Pull request merged**